src/web/statics.rs

```diff
@@ -28,15 +28,21 @@ pub(crate) fn static_handler(req: &mut Request) -> IronResult<Response> {
 }
 
 fn serve_file(req: &Request, file: &str) -> IronResult<Response> {
-    // Filter out files that attempt to traverse directories
-    if file.contains("..") || file.contains('/') || file.contains('\\') {
-        return Err(Nope::ResourceNotFound.into());
-    }
-
     // Find the first path that actually exists
     let path = STATIC_SEARCH_PATHS
         .iter()
-        .map(|p| Path::new(p).join(file))
+        .filter_map(|root| {
+            let path = Path::new(root).join(file);
+
+            // Prevent accessing static files outside the root. This could happen if the path
+            // contains `/` or `..`. The check doesn't outright prevent those strings to be present
+            // to allow accessing files in subdirectories.
```

Comment:
Are there tests for this?

Reply:
Well, it turns out there *is already* a test about this (`directory_traversal`), but the test itself was broken :sweat_smile: 

Fixed both my code (I forgot to canonicalize) and the test itself.

Reply:
> Fixed both my code (I forgot to canonicalize) and the test itself.

Did you forget to push your changes?

Reply:
Woops, yes, pushed!